File: Implement strStr()/Implement strStr().java
public class Solution {
    public String strStr(String haystack, String needle) {
        if(haystack == null || needle == null) return null;
        else if (needle.length() == 0) return haystack;
        for(int i = 0; i < haystack.length() - needle.length() + 1; i++) {
            for(int j = 0; j < needle.length(); j++) {
                if(haystack.charAt(i + j) != needle.charAt(j)) break;
                if(j == needle.length() - 1) return haystack.substring(i);
            }
        }
        return null;
    }
}

File: Implement strStr()/Implement strStr().cpp
class Solution {
public:
    char *strStr(char *haystack, char *needle) {
        if (*needle == '\0') return haystack;
        while (*haystack != '\0') {
            if (*haystack == *needle) {
                char *p = needle;
                char *start = haystack;
                while (*haystack == *p && *haystack != '\0') {
                    ++haystack;
                    ++p;
                }
                if (*p == '\0') return start;
                else if (*haystack == '\0') return nullptr;
                else haystack = start + 1;
            }
            else ++haystack;
        }
        return nullptr;
    }
};